File: Step004_SMSIf_NestedConditions/IfConditions_Nested.m
%**************************************************************
%* AceGen    7.505 Linux (16 Aug 22)                          *
%*           Co. J. Korelc  2020           29 May 24 16:47:19 *
%**************************************************************
% User     : Full professional version
% Notebook : AceGen-SMSIf_Nested
% Evaluation time                 : 1 s     Mode  : Optimal
% Number of formulae              : 47      Method: Automatic
% Subroutine                      : IfConditions_Nested size: 287
% Total size of Mathematica  code : 287 subexpressions
% Total size of Matlab code      : 1297 bytes 

%*********************** F U N C T I O N **************************
function[y1Output,Dy1DxOutput,y2Output,Dy2DxOutput,y3Output,Dy3DxOutput,y4Output,Dy4DxOutput]=IfConditions_Nested(xInput);
persistent v;
if size(v)<189
  v=zeros(189,'double');
end;
v(1)=xInput;
b5=v(1)<-10e0;
b2=v(1)>=0;
if(b2)
 v(27)=-(1/Power(v(1),2))+2e0*v(1)+cos(v(1));
 v(9)=25e0+1/v(1)+(v(1)*v(1))+sin(v(1));
 v(28)=v(27);
 v(4)=v(9);
else;
 if(b5)
  v(42)=sqrt(-v(1));
  v(29)=-9e0-1/(2e0*v(42))+cos(v(1));
  v(12)=-9e0*v(1)+v(42)+sin(v(1));
  v(30)=v(29);
  v(7)=v(12);
 else;
  v(31)=2e0*v(1);
  v(13)=(v(1)*v(1));
  v(30)=v(31);
  v(7)=v(13);
 end;
 v(28)=v(30);
 v(4)=v(7);
end;
if(b2)
 v(34)=v(27);
 v(10)=v(9);
else;
 if(b5)
  v(34)=v(29);
  v(10)=v(12);
 else;
  v(34)=v(31);
  v(10)=v(13);
 end;
end;
v(37)=0;
v(14)=0;
if(b2)
 v(37)=v(27);
 v(14)=v(9);
else;
 if(b5)
  v(37)=v(29);
  v(14)=v(12);
 else;
  v(37)=v(31);
  v(14)=v(13);
 end;
end;
if(b2)
 v(40)=v(27);
 v(22)=v(9);
else;
end;
if(b5)
 v(43)=sqrt(-v(1));
 v(40)=-9e0-1/(2e0*v(43))+cos(v(1));
 v(22)=-9e0*v(1)+v(43)+sin(v(1));
else;
end;
if(v(1)<0 && v(1)>=-10e0)
 v(40)=2e0*v(1);
 v(22)=(v(1)*v(1));
else;
end;
y1Output=v(4);
Dy1DxOutput=v(28);
y2Output=v(10);
Dy2DxOutput=v(34);
y3Output=v(14);
Dy3DxOutput=v(37);
y4Output=v(22);
Dy4DxOutput=v(40);


function [x]=SMSKDelta(i,j)
if (i==j) , x=1; else x=0; end;
end
function [x]=SMSDeltaPart(a,i,j,k)
l=round(i/j);
if (mod(i,j) ~= 0 | l>k) , x=0; else x=a(l); end;
end
function [x]=Power(a,b)
x=a^b;
end
function [x]=SMSTernaryOperator(a,b,c)
if (c) , x=a; else x=b; end;
end
end

File: Step004_SMSIf_NestedConditions/IfConditions_Nested.f
!**************************************************************
!* AceGen    7.505 Linux (16 Aug 22)                          *
!*           Co. J. Korelc  2020           24 Apr 24 17:52:13 *
!**************************************************************
! User     : Full professional version
! Notebook : AceGen-SMSIf_Nested
! Evaluation time                 : 1 s     Mode  : Optimal
! Number of formulae              : 27      Method: Automatic
! Subroutine                      : IfConditions_Nested size: 179
! Total size of Mathematica  code : 179 subexpressions
! Total size of Fortran code      : 1114 bytes

!******************* S U B R O U T I N E **********************
      SUBROUTINE IfConditions_Nested(xInput,y1Output,Dy1DxOutput
     &,y2Output,Dy2DxOutput,y3Output,Dy3DxOutput)
      IMPLICIT NONE
      include 'sms.h'
      LOGICAL b2,b3,b6,b10,b16
      DOUBLE PRECISION v(147),xInput,y1Output,Dy1DxOutput
     &,y2Output,Dy2DxOutput,y3Output,Dy3DxOutput
      v(1)=xInput
      v(13)=2d0*v(1)
      v(5)=(v(1)*v(1))
      b2=v(1).ge.0d0
      IF(b2) THEN
       v(14)=-(1d0/v(1)**2)+v(13)+dcos(v(1))
       v(7)=25d0+1d0/v(1)+v(5)+dsin(v(1))
       v(15)=v(14)
       v(4)=v(7)
      ELSE
       b16=v(1).lt.(-10d0)
       v(15)=MERGE(DBLE(-99),v(13),b16)
       v(4)=MERGE((-99d0)*v(1),v(5),b16)
      ENDIF
      IF(b2) THEN
       v(19)=v(14)
       v(8)=v(7)
      ELSE
       v(19)=v(13)
       v(8)=v(5)
      ENDIF
      v(22)=0d0
      v(9)=0d0
      IF(b2) THEN
       v(22)=v(14)
       v(9)=v(7)
      ELSE
       v(22)=v(13)
       v(9)=v(5)
      ENDIF
      y1Output=v(4)
      Dy1DxOutput=v(15)
      y2Output=v(8)
      Dy2DxOutput=v(19)
      y3Output=v(9)
      Dy3DxOutput=v(22)
      END
